Tests/UnitTests/Misc/SigningTests.swift

```diff
@@ -68,76 +68,151 @@ class SigningTests: TestCase {
 
         let message = "Hello World"
         let nonce = "nonce"
+        let requestTime = 1677005916012
         let signature = "this is not a signature"
 
-        expect(Signing.verify(message: message.asData,
-                              nonce: nonce.asData,
-                              hasValidSignature: signature,
-                              with: Signing.loadPublicKey())) == false
+        expect(Signing.verify(
+            signature: signature,
+            with: .init(
+                notModifiedResponse: false,
+                message: message.asData,
+                nonce: nonce.asData,
+                requestTime: requestTime,
+                eTag: nil
+            ),
+            publicKey: Signing.loadPublicKey()
+        )) == false
 
         logger.verifyMessageWasLogged("Signature is not base64: \(signature)")
     }
 
     func testVerifySignatureWithInvalidSignature() throws {
-        expect(Signing.verify(message: "Hello World".asData,
-                              nonce: "nonce".asData,
-                              hasValidSignature: "invalid signature".asData.base64EncodedString(),
-                              with: Signing.loadPublicKey())) == false
+        expect(Signing.verify(
+            signature: "invalid signature".asData.base64EncodedString(),
+            with: .init(
+                notModifiedResponse: false,
+                message: "Hello World".asData,
+                nonce: "nonce".asData,
+                requestTime: 1677005916012,
+                eTag: nil
+            ),
+            publicKey: Signing.loadPublicKey()
+        )) == false
     }
 
     func testVerifySignatureLogsWarningWhenFail() throws {
         let logger = TestLogHandler()
 
-        _ = Signing.verify(message: "Hello World".asData,
-                           nonce: "nonce".asData,
-                           hasValidSignature: "invalid signature".asData.base64EncodedString(),
-                           with: Signing.loadPublicKey())
+        _ = Signing.verify(signature: "invalid signature".asData.base64EncodedString(),
+                           with: .init(
+                            notModifiedResponse: false,
+                            message: "Hello World".asData,
+                            nonce: "nonce".asData,
+                            requestTime: 1677005916012,
+                            eTag: nil
```

Comment:
Include etag for the text. 200 responses will do have etag field.

Reply:
No longer relevant with the other refactor 👍🏻